src/renderer_gl.cpp

```diff
@@ -5396,7 +5396,8 @@ namespace bgfx { namespace gl
 					GL_CHECK(glVertexAttribDivisor(loc, 0) );
 
 					uint32_t baseVertex = _baseVertex*_layout.m_stride + _layout.m_offset[attr];
-					if ( (BX_ENABLED(BGFX_CONFIG_RENDERER_OPENGL >= 30) || s_renderGL->m_gles3)
+					// Use glVertexAttribIPointer on opengl only. Some Android devices like Samsung S22 do not support this function properly on opengles.
+					if ( (BX_ENABLED(BGFX_CONFIG_RENDERER_OPENGL) && s_renderGL->m_gles3) 
```

Comment:
This logic is non-sense... you're testing if it's OpenGL and GLES3 which is never true.

Reply:
Also it's part of GLES3:
https://github.com/bkaradzic/bgfx/blob/8d412e16e40e8d55bcb6ec3cedf18e39711dfd8e/3rdparty/khronos/GLES3/gl3.h#L1034

Reply:
Oops. You're right. My pull request makes no sense. 

Although glVertexAttribIPointer is part of Opengles 3.0 I've encountered some devices (like the Samsung S22 and the Chrome browser) that do not support this function. Therefore I've deactivated this function for my project.

I changed my code to:
if ( (BX_ENABLED(BGFX_CONFIG_RENDERER_OPENGL >= 30))

This is working fine for my project. But I cannot say if this would have a side effect to other projects.

Reply:
Yeah, keep your change local.

"Samsung S22" needs to fix their driver.